### PR TITLE
[7.x Backport] Fix error in instantiating bad response code exceptions

### PR DESCRIPTION
**Pull request by andrewvc**
Backport of https://github.com/logstash-plugins/logstash-output-elasticsearch/pull/637#issuecomment-330187367 to 7.x

### CODE REVIEW

**Comment by jordansissel** [comment]
https://travis-ci.org/logstash-plugins/logstash-output-elasticsearch/jobs/278255908

12 test failures against ES 6.0.0-beta2 (version[6.0.0-beta2-SNAPSHOT))

rspec ./spec/integration/outputs/painless_update_spec.rb:36 # Update actions using painless scripts scripted updates should increment a counter with event/doc 'count' variable

rspec ./spec/integration/outputs/painless_update_spec.rb:93 # Update actions using painless scripts scripted updates should increment a counter with event/doc 'count' variable with indexed script

rspec ./spec/integration/outputs/painless_update_spec.rb:44 # Update actions using painless scripts scripted updates should increment a counter with event/doc '[data][count]' nested variable

rspec ./spec/integration/outputs/painless_update_spec.rb:132 # Update actions using painless scripts updates with scripted upsert should create new documents with upsert content

rspec ./spec/integration/outputs/painless_update_spec.rb:140 # Update actions using painless scripts updates with scripted upsert should create new documents with event/doc as script params

rspec ./spec/integration/outputs/groovy_update_spec.rb:140 # Update actions using groovy scripts updates with scripted upsert should create new documents with event/doc as script params

rspec ./spec/integration/outputs/groovy_update_spec.rb:132 # Update actions using groovy scripts updates with scripted upsert should create new documents with upsert content

rspec ./spec/integration/outputs/groovy_update_spec.rb:52 # Update actions using groovy scripts scripted updates should increment a counter with event/doc 'count' variable with inline script

rspec ./spec/integration/outputs/groovy_update_spec.rb:93 # Update actions using groovy scripts scripted updates should increment a counter with event/doc 'count' variable with indexed script

rspec ./spec/integration/outputs/groovy_update_spec.rb:44 # Update actions using groovy scripts scripted updates should increment a counter with event/doc '[data][count]' nested variable

rspec ./spec/integration/outputs/groovy_update_spec.rb:36 # Update actions using groovy scripts scripted updates should increment a counter with event/doc 'count' variable

rspec ./spec/integration/outputs/groovy_update_spec.rb:65 # Update actions using groovy scripts scripted updates should increment a counter with event/doc 'count' variable with event/doc as upsert and inline script

I feel like these are unrelated to your test (6.0 has no groovy language, iirc, so these *should* be failing and are unrelated to your change?)

**Comment by andrewvc** [comment]
@jordansissel yes, unrelated. I believe we may have a missing patch WRT scripting pending for the 7.x branch.

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +11,19 @@
...
+
+## 8.0.1
+  - Fix some documentation issues
+
+## 8.0.0
+ - Breaking: make deprecated options :flush_size and :idle_flush_time obsolete
+ - Remove obsolete options :max_retries and :retry_max_items
+ - Fix: handling of initial single big event
+ - Fix: typo was enabling http compression by default this returns it back to false
+>>>>>>> a4e18a8... Fix error in instantiating bad response code exceptions
```
Comment (jsvd) [inline CHANGELOG.md]: these merge leftovers must be removed

**Comment by jsvd** [comment]
closed in favor of #685

**Pull request closed without merging**